MDANSE/Src/MDANSE/Framework/AtomSelector/selector.py

```
@@ -13,362 +13,170 @@
-        return settings
-
-    def load_from_json(self, json_string: str) -> None:
-        """Load the selection settings from a JSON string.
-
-        Parameters
-        ----------
-        json_string : str
-            The JSON string of settings.
-        """
-        self.update_settings(self.json_to_settings(json_string), reset_first=True)
-
-    def check_valid_setting(self, settings: dict[str, Union[bool, dict]]) -> bool:
-        """Checks that the input settings are valid.
-
-        Parameters
-        ----------
-        settings : dict[str, bool | dict]
-            The selection settings.
-
-        Returns
-        -------
-        bool
-            True if settings are valid.
-        """
-        setting_keys = self._default.keys()
-        dict_setting_keys = self._kwarg_keys.keys()
-        for k0, v0 in settings.items():
-            if k0 not in setting_keys:
-                return False
-
-            if k0 not in dict_setting_keys:
-                if not isinstance(v0, bool):
-                    return False
-
-            if k0 in dict_setting_keys:
-                if not isinstance(v0, dict):
-                    return False
-                for k1, v1 in v0.items():
-                    if k1 not in self._kwarg_vals[k0]:
-                        return False
-                    if not isinstance(v1, bool):
-                        return False
-
-        return True
-
-    def check_valid_json_settings(self, json_string: str) -> bool:
-        """Checks that the input JSON setting string is valid.
-
-        Parameters
-        ----------
-        json_string : str
-            The JSON string of settings.
-
-        Returns
-        -------
-        bool
-            True if settings are valid.
-        """
-        try:
-            settings = self.json_to_settings(json_string)
-        except ValueError:
-            return False
-        return self.check_valid_setting(settings)
+            if isinstance(v0, dict):
```

If it's not a `dict` it's silently ignored? Are we sure that's the behaviour we want?